libpysal/weights/weights.py

```diff
@@ -849,11 +852,11 @@ def histogram(self):
         value is the number of neighbors for that unit.
         """
         if "histogram" not in self._cache:
-            ct, bin = np.histogram(
+            ct, bin_ = np.histogram(
                 list(self.cardinalities.values()),
                 list(range(self.min_neighbors, self.max_neighbors + 2)),
             )
-            self._histogram = list(zip(bin, ct))
+            self._histogram = list(zip(bin_, ct, strict=False))
```

Comment:
```suggestion
            self._histogram = list(zip(bin_[:-1], ct, strict=True))
```
It is relying on `zip` omitting the last bin edge. We should do that explicitly.